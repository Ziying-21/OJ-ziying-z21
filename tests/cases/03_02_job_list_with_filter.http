###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":0,"language":"Rust","problem_id":1,"source_code":"fn main() {let mut line1 = String::new();std::io::stdin().read_line(&mut line1).unwrap();let a: i32 = line1.trim().parse().unwrap();let mut line2 = String::new();std::io::stdin().read_line(&mut line2).unwrap();let b: i32 = line2.trim().parse().unwrap();println!(\"{}\", a + b);}","user_id":0}
###
# Got response:
# HTTP 200
# content-length: 1137
# content-type: application/json
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:07 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Waiting","time":0},{"id":1,"info":"","memory":0,"result":"Waiting","time":0},{"id":2,"info":"","memory":0,"result":"Waiting","time":0},{"id":3,"info":"","memory":0,"result":"Waiting","time":0},{"id":4,"info":"","memory":0,"result":"Waiting","time":0},{"id":5,"info":"","memory":0,"result":"Waiting","time":0},{"id":6,"info":"","memory":0,"result":"Waiting","time":0},{"id":7,"info":"","memory":0,"result":"Waiting","time":0},{"id":8,"info":"","memory":0,"result":"Waiting","time":0},{"id":9,"info":"","memory":0,"result":"Waiting","time":0},{"id":10,"info":"","memory":0,"result":"Waiting","time":0}],"created_time":"2022-09-09T02:19:08.042Z","id":0,"result":"Waiting","score":0.0,"state":"Queueing","submission":{"contest_id":0,"language":"Rust","problem_id":1,"source_code":"fn main() {let mut line1 = String::new();std::io::stdin().read_line(&mut line1).unwrap();let a: i32 = line1.trim().parse().unwrap();let mut line2 = String::new();std::io::stdin().read_line(&mut line2).unwrap();let b: i32 = line2.trim().parse().unwrap();println!(\"{}\", a + b);}","user_id":0},"updated_time":""}
###
# Send request:
GET http://127.0.0.1:12345/jobs/0 HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 1222
# content-type: application/json
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:08 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Compilation Success","time":164831},{"id":1,"info":"","memory":0,"result":"Accepted","time":61218},{"id":2,"info":"","memory":0,"result":"Accepted","time":1556},{"id":3,"info":"","memory":0,"result":"Accepted","time":1870},{"id":4,"info":"","memory":0,"result":"Accepted","time":1856},{"id":5,"info":"","memory":0,"result":"Accepted","time":1917},{"id":6,"info":"","memory":0,"result":"Accepted","time":2014},{"id":7,"info":"","memory":0,"result":"Accepted","time":1479},{"id":8,"info":"","memory":0,"result":"Accepted","time":1444},{"id":9,"info":"","memory":0,"result":"Accepted","time":1543},{"id":10,"info":"","memory":0,"result":"Accepted","time":1485}],"created_time":"2022-09-09T02:19:08.042Z","id":0,"result":"Accepted","score":100.0,"state":"Finished","submission":{"contest_id":0,"language":"Rust","problem_id":1,"source_code":"fn main() {let mut line1 = String::new();std::io::stdin().read_line(&mut line1).unwrap();let a: i32 = line1.trim().parse().unwrap();let mut line2 = String::new();std::io::stdin().read_line(&mut line2).unwrap();let b: i32 = line2.trim().parse().unwrap();println!(\"{}\", a + b);}","user_id":0},"updated_time":"2022-09-09T02:19:08.298Z"}
###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":0,"language":"Rust","problem_id":2,"source_code":"fn main() {let mut data: Vec<u32> = Vec::new();loop {let mut line = String::new();if std::io::stdin().read_line(&mut line).unwrap() == 0 {break;}data.push(line.trim().parse().unwrap());}data.sort_by_key(|&x| x);data.into_iter().for_each(|x| println!(\"{}\", x));}","user_id":0}
###
# Got response:
# HTTP 200
# content-length: 1122
# content-type: application/json
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:08 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Waiting","time":0},{"id":1,"info":"","memory":0,"result":"Waiting","time":0},{"id":2,"info":"","memory":0,"result":"Waiting","time":0},{"id":3,"info":"","memory":0,"result":"Waiting","time":0},{"id":4,"info":"","memory":0,"result":"Waiting","time":0},{"id":5,"info":"","memory":0,"result":"Waiting","time":0},{"id":6,"info":"","memory":0,"result":"Waiting","time":0},{"id":7,"info":"","memory":0,"result":"Waiting","time":0},{"id":8,"info":"","memory":0,"result":"Waiting","time":0},{"id":9,"info":"","memory":0,"result":"Waiting","time":0},{"id":10,"info":"","memory":0,"result":"Waiting","time":0}],"created_time":"2022-09-09T02:19:09.056Z","id":1,"result":"Waiting","score":0.0,"state":"Queueing","submission":{"contest_id":0,"language":"Rust","problem_id":2,"source_code":"fn main() {let mut data: Vec<u32> = Vec::new();loop {let mut line = String::new();if std::io::stdin().read_line(&mut line).unwrap() == 0 {break;}data.push(line.trim().parse().unwrap());}data.sort_by_key(|&x| x);data.into_iter().for_each(|x| println!(\"{}\", x));}","user_id":0},"updated_time":""}
###
# Send request:
GET http://127.0.0.1:12345/jobs/1 HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 1207
# content-type: application/json
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:10 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Compilation Success","time":205930},{"id":1,"info":"","memory":0,"result":"Accepted","time":65894},{"id":2,"info":"","memory":0,"result":"Accepted","time":1863},{"id":3,"info":"","memory":0,"result":"Accepted","time":2012},{"id":4,"info":"","memory":0,"result":"Accepted","time":1895},{"id":5,"info":"","memory":0,"result":"Accepted","time":2160},{"id":6,"info":"","memory":0,"result":"Accepted","time":1856},{"id":7,"info":"","memory":0,"result":"Accepted","time":1934},{"id":8,"info":"","memory":0,"result":"Accepted","time":1813},{"id":9,"info":"","memory":0,"result":"Accepted","time":2233},{"id":10,"info":"","memory":0,"result":"Accepted","time":1900}],"created_time":"2022-09-09T02:19:09.056Z","id":1,"result":"Accepted","score":100.0,"state":"Finished","submission":{"contest_id":0,"language":"Rust","problem_id":2,"source_code":"fn main() {let mut data: Vec<u32> = Vec::new();loop {let mut line = String::new();if std::io::stdin().read_line(&mut line).unwrap() == 0 {break;}data.push(line.trim().parse().unwrap());}data.sort_by_key(|&x| x);data.into_iter().for_each(|x| println!(\"{}\", x));}","user_id":0},"updated_time":"2022-09-09T02:19:09.364Z"}
###
# Send request:
POST http://127.0.0.1:12345/jobs HTTP/1.1
content-type: application/json

{"contest_id":0,"language":"C","problem_id":0,"source_code":"#include <stdio.h>\nint main() { printf(\"Hello World!\\n\"); return 0; }","user_id":0}
###
# Got response:
# HTTP 200
# content-length: 406
# content-type: application/json
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:10 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Waiting","time":0},{"id":1,"info":"","memory":0,"result":"Waiting","time":0}],"created_time":"2022-09-09T02:19:10.075Z","id":2,"result":"Waiting","score":0.0,"state":"Queueing","submission":{"contest_id":0,"language":"C","problem_id":0,"source_code":"#include <stdio.h>\nint main() { printf(\"Hello World!\\n\"); return 0; }","user_id":0},"updated_time":""}
###
# Send request:
GET http://127.0.0.1:12345/jobs/2 HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 454
# content-type: application/json
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:11 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Compilation Success","time":60650},{"id":1,"info":"","memory":0,"result":"Accepted","time":62734}],"created_time":"2022-09-09T02:19:10.075Z","id":2,"result":"Accepted","score":100.0,"state":"Finished","submission":{"contest_id":0,"language":"C","problem_id":0,"source_code":"#include <stdio.h>\nint main() { printf(\"Hello World!\\n\"); return 0; }","user_id":0},"updated_time":"2022-09-09T02:19:10.204Z"}
###
# Send request:
GET http://127.0.0.1:12345/jobs?result=Accepted HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 2887
# content-type: application/json
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:11 GMT
# 
# [{"cases":[{"id":0,"info":"","memory":0,"result":"Compilation Success","time":164831},{"id":1,"info":"","memory":0,"result":"Accepted","time":61218},{"id":2,"info":"","memory":0,"result":"Accepted","time":1556},{"id":3,"info":"","memory":0,"result":"Accepted","time":1870},{"id":4,"info":"","memory":0,"result":"Accepted","time":1856},{"id":5,"info":"","memory":0,"result":"Accepted","time":1917},{"id":6,"info":"","memory":0,"result":"Accepted","time":2014},{"id":7,"info":"","memory":0,"result":"Accepted","time":1479},{"id":8,"info":"","memory":0,"result":"Accepted","time":1444},{"id":9,"info":"","memory":0,"result":"Accepted","time":1543},{"id":10,"info":"","memory":0,"result":"Accepted","time":1485}],"created_time":"2022-09-09T02:19:08.042Z","id":0,"result":"Accepted","score":100.0,"state":"Finished","submission":{"contest_id":0,"language":"Rust","problem_id":1,"source_code":"fn main() {let mut line1 = String::new();std::io::stdin().read_line(&mut line1).unwrap();let a: i32 = line1.trim().parse().unwrap();let mut line2 = String::new();std::io::stdin().read_line(&mut line2).unwrap();let b: i32 = line2.trim().parse().unwrap();println!(\"{}\", a + b);}","user_id":0},"updated_time":"2022-09-09T02:19:08.298Z"},{"cases":[{"id":0,"info":"","memory":0,"result":"Compilation Success","time":205930},{"id":1,"info":"","memory":0,"result":"Accepted","time":65894},{"id":2,"info":"","memory":0,"result":"Accepted","time":1863},{"id":3,"info":"","memory":0,"result":"Accepted","time":2012},{"id":4,"info":"","memory":0,"result":"Accepted","time":1895},{"id":5,"info":"","memory":0,"result":"Accepted","time":2160},{"id":6,"info":"","memory":0,"result":"Accepted","time":1856},{"id":7,"info":"","memory":0,"result":"Accepted","time":1934},{"id":8,"info":"","memory":0,"result":"Accepted","time":1813},{"id":9,"info":"","memory":0,"result":"Accepted","time":2233},{"id":10,"info":"","memory":0,"result":"Accepted","time":1900}],"created_time":"2022-09-09T02:19:09.056Z","id":1,"result":"Accepted","score":100.0,"state":"Finished","submission":{"contest_id":0,"language":"Rust","problem_id":2,"source_code":"fn main() {let mut data: Vec<u32> = Vec::new();loop {let mut line = String::new();if std::io::stdin().read_line(&mut line).unwrap() == 0 {break;}data.push(line.trim().parse().unwrap());}data.sort_by_key(|&x| x);data.into_iter().for_each(|x| println!(\"{}\", x));}","user_id":0},"updated_time":"2022-09-09T02:19:09.364Z"},{"cases":[{"id":0,"info":"","memory":0,"result":"Compilation Success","time":60650},{"id":1,"info":"","memory":0,"result":"Accepted","time":62734}],"created_time":"2022-09-09T02:19:10.075Z","id":2,"result":"Accepted","score":100.0,"state":"Finished","submission":{"contest_id":0,"language":"C","problem_id":0,"source_code":"#include <stdio.h>\nint main() { printf(\"Hello World!\\n\"); return 0; }","user_id":0},"updated_time":"2022-09-09T02:19:10.204Z"}]
###
# Send request:
GET http://127.0.0.1:12345/jobs?problem_id=0&state=Finished HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 456
# content-type: application/json
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:11 GMT
# 
# [{"cases":[{"id":0,"info":"","memory":0,"result":"Compilation Success","time":60650},{"id":1,"info":"","memory":0,"result":"Accepted","time":62734}],"created_time":"2022-09-09T02:19:10.075Z","id":2,"result":"Accepted","score":100.0,"state":"Finished","submission":{"contest_id":0,"language":"C","problem_id":0,"source_code":"#include <stdio.h>\nint main() { printf(\"Hello World!\\n\"); return 0; }","user_id":0},"updated_time":"2022-09-09T02:19:10.204Z"}]
###
# Send request:
GET http://127.0.0.1:12345/jobs?problem_id=0&state=Finished HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 456
# content-type: application/json
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:11 GMT
# 
# [{"cases":[{"id":0,"info":"","memory":0,"result":"Compilation Success","time":60650},{"id":1,"info":"","memory":0,"result":"Accepted","time":62734}],"created_time":"2022-09-09T02:19:10.075Z","id":2,"result":"Accepted","score":100.0,"state":"Finished","submission":{"contest_id":0,"language":"C","problem_id":0,"source_code":"#include <stdio.h>\nint main() { printf(\"Hello World!\\n\"); return 0; }","user_id":0},"updated_time":"2022-09-09T02:19:10.204Z"}]
###
# Send request:
GET http://127.0.0.1:12345/jobs?language=C HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 456
# content-type: application/json
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:11 GMT
# 
# [{"cases":[{"id":0,"info":"","memory":0,"result":"Compilation Success","time":60650},{"id":1,"info":"","memory":0,"result":"Accepted","time":62734}],"created_time":"2022-09-09T02:19:10.075Z","id":2,"result":"Accepted","score":100.0,"state":"Finished","submission":{"contest_id":0,"language":"C","problem_id":0,"source_code":"#include <stdio.h>\nint main() { printf(\"Hello World!\\n\"); return 0; }","user_id":0},"updated_time":"2022-09-09T02:19:10.204Z"}]
###
# Send request:
GET http://127.0.0.1:12345/jobs/2 HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 454
# content-type: application/json
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:11 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Compilation Success","time":60650},{"id":1,"info":"","memory":0,"result":"Accepted","time":62734}],"created_time":"2022-09-09T02:19:10.075Z","id":2,"result":"Accepted","score":100.0,"state":"Finished","submission":{"contest_id":0,"language":"C","problem_id":0,"source_code":"#include <stdio.h>\nint main() { printf(\"Hello World!\\n\"); return 0; }","user_id":0},"updated_time":"2022-09-09T02:19:10.204Z"}
###
# Send request:
GET http://127.0.0.1:12345/jobs/0 HTTP/1.1


###
# Got response:
# HTTP 200
# content-length: 1222
# content-type: application/json
# access-control-allow-credentials: true
# vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
# access-control-expose-headers: content-type
# date: Fri, 09 Sep 2022 02:19:11 GMT
# 
# {"cases":[{"id":0,"info":"","memory":0,"result":"Compilation Success","time":164831},{"id":1,"info":"","memory":0,"result":"Accepted","time":61218},{"id":2,"info":"","memory":0,"result":"Accepted","time":1556},{"id":3,"info":"","memory":0,"result":"Accepted","time":1870},{"id":4,"info":"","memory":0,"result":"Accepted","time":1856},{"id":5,"info":"","memory":0,"result":"Accepted","time":1917},{"id":6,"info":"","memory":0,"result":"Accepted","time":2014},{"id":7,"info":"","memory":0,"result":"Accepted","time":1479},{"id":8,"info":"","memory":0,"result":"Accepted","time":1444},{"id":9,"info":"","memory":0,"result":"Accepted","time":1543},{"id":10,"info":"","memory":0,"result":"Accepted","time":1485}],"created_time":"2022-09-09T02:19:08.042Z","id":0,"result":"Accepted","score":100.0,"state":"Finished","submission":{"contest_id":0,"language":"Rust","problem_id":1,"source_code":"fn main() {let mut line1 = String::new();std::io::stdin().read_line(&mut line1).unwrap();let a: i32 = line1.trim().parse().unwrap();let mut line2 = String::new();std::io::stdin().read_line(&mut line2).unwrap();let b: i32 = line2.trim().parse().unwrap();println!(\"{}\", a + b);}","user_id":0},"updated_time":"2022-09-09T02:19:08.298Z"}
