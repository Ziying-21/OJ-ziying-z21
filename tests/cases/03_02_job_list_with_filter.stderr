[2022-09-09T02:19:07Z INFO  actix_server::builder] Starting 8 workers
[2022-09-09T02:19:07Z INFO  actix_server::server] Actix runtime found; starting in Actix runtime
[2022-09-09T02:19:08Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET / HTTP/1.1" 404 0 "-" "-" 0.000197
[2022-09-09T02:19:08Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 200 1137 "-" "-" 0.000661
[2022-09-09T02:19:09Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /jobs/0 HTTP/1.1" 200 1222 "-" "-" 0.000363
[2022-09-09T02:19:09Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 200 1122 "-" "-" 0.001060
[2022-09-09T02:19:10Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /jobs/1 HTTP/1.1" 200 1207 "-" "-" 0.000711
[2022-09-09T02:19:10Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /jobs HTTP/1.1" 200 406 "-" "-" 0.002289
[2022-09-09T02:19:11Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /jobs/2 HTTP/1.1" 200 454 "-" "-" 0.000281
[2022-09-09T02:19:11Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /jobs?result=Accepted HTTP/1.1" 200 2887 "-" "-" 0.000235
[2022-09-09T02:19:11Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /jobs?problem_id=0&state=Finished HTTP/1.1" 200 456 "-" "-" 0.000105
[2022-09-09T02:19:11Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /jobs?problem_id=0&state=Finished HTTP/1.1" 200 456 "-" "-" 0.000098
[2022-09-09T02:19:11Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /jobs?language=C HTTP/1.1" 200 456 "-" "-" 0.000096
[2022-09-09T02:19:11Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /jobs/2 HTTP/1.1" 200 454 "-" "-" 0.000108
[2022-09-09T02:19:11Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /jobs/0 HTTP/1.1" 200 1222 "-" "-" 0.000131
